pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,539 @@
+			{Model: APIParamModelInt, Name: "limit"},
+		},
+	},
+	{
+		ID:        "pd_region_check",
+		Component: model.NodeKindPD,
+		Path:      "/regions/check/{state}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{
+				Model: APIParamModelEnum([]EnumItem{
+					{Value: "miss-peer"},
+					{Value: "extra-peer"},
+					{Value: "down-peer"},
+					{Value: "pending-peer"},
+					{Value: "offline-peer"},
+					{Value: "empty-peer"},
+					{Value: "hist-peer"},
+					{Value: "hist-keys"},
+				}),
+				Name: "state", Required: true},
+		},
+		QueryParams: []*endpoint.APIParam{
+			{Model: APIParamModelInt, Name: "bound"},
+		},
+		OnResolve: func(resolvedPayload *endpoint.ResolvedRequestPayload) error {
+			state := resolvedPayload.PathParams["state"]
+			val := resolvedPayload.QueryParams.Get("bound")
+
+			if val == "" {
+				if strings.EqualFold(state, "hist-size") {
+					resolvedPayload.QueryParams.Set("bound", "10")
+				} else if strings.EqualFold(state, "hist-keys") {
+					resolvedPayload.QueryParams.Set("bound", "10000")
+				}
+			}
+			return nil
+		},
+	},
+	{
+		ID:        "pd_scheduler_show",
+		Component: model.NodeKindPD,
+		Path:      "/schedulers",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: APIParamModelText, Name: "status"},
+		},
+	},
+	{
+		ID:        "pd_stores",
+		Component: model.NodeKindPD,
+		Path:      "/stores",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: APIParamModelMultiValue, Name: "state"},
+		},
+		OnResolve: func(resolvedPayload *endpoint.ResolvedRequestPayload) error {
+			vals := resolvedPayload.QueryParams["state"]
+			resolvedPayload.QueryParams.Del("state")
+			if len(vals) != 0 {
+				for _, state := range vals {
+					stateValue, ok := metapb.StoreState_value[state]
+					if !ok {
+						return fmt.Errorf("unknown state: %s", state)
+					}
+					resolvedPayload.QueryParams.Add("state", strconv.Itoa(int(stateValue)))
+				}
+			}
+			return nil
+		},
+	},
+	{
+		ID:        "pd_store_id",
+		Component: model.NodeKindPD,
+		Path:      "/store/{storeID}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: APIParamModelText, Name: "storeID", Required: true},
+		},
+	},
+	{
+		ID:        "pd_pprof",
+		Component: model.NodeKindPD,
+		Path:      "/debug/pprof/{kind}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			pprofKindsParam,
+		},
+		QueryParams: []*endpoint.APIParam{
+			pprofSecondsParam,
+			pprofDebugParam,
+		},
+		OnResolve: func(resolvedPayload *endpoint.ResolvedRequestPayload) error {
+			if err := timeoutMiddleware(resolvedPayload, resolvedPayload.QueryParams.Get("seconds")); err != nil {
+				return err
+			}
+			return nil
+		},
+	},
+	// tikv
+	{
+		ID:        "tikv_config",
+		Component: model.NodeKindTiKV,
+		Path:      "/config",
+		Method:    endpoint.MethodGet,
+	},
+	// TODO: open after http client refactor
```

Comment:
Why?

Reply:
Here we need http client support to add the header to fetch the pprof data:
`resolvedPayload.Header.Add("Content-Type", "application/protobuf")`

or we need to support override response header to fetch the svg data.:
`xxx.Response.Header.Set("Content-Type", "image/svg+xml")`

Reply:
What will happen if we do nothing?

Reply:
we will get svg data without the ext?

Reply:
interesting.. Does TiKV not respond a correct content-type?

Reply:
Yeah, maybe you can help me take a look at this pr~ https://github.com/tikv/tikv/pull/10951